sw/banshee/tests/ssr_inst_check.c

```diff
@@ -40,17 +40,16 @@ int main() {
         "3:\n"
 
         // ------------------------------------
-        // write and read to ssr1 stride3
+        // write and read to ssr1 stride2
         // ------------------------------------
-        "li     t0, (0x100 | 0x48)\n"  // address
-        "li     t1, 0x98765432\n"      // data
-        "scfgw  t1, t0\n"              // val, adr
-        "scfgr  t3, t0\n"              // val, adr
+        "li     t0, (0x1 | 8<<5)\n"  // address
+        "li     t1, 0x98765432\n"    // data
+        "scfgw  t1, t0\n"            // val, adr
+        "scfgr  t3, t0\n"            // val, adr
         // compare
         "beq    t1, t3, 4f\n"
         "addi   %[errs], %[errs], 1\n"  // increment error on mismatch
         "4:\n"
-        : [ errs ] "+r"(errs)
-        :);
+        : [ errs ] "+r"(errs)::"t0", "t1", "t2", "t3");
```

Comment:
Very nice. Could you also add some tests mixing the immediate and register versions to make sure that both use the same scheme to calculate the address?

Reply:
Good idea, will do that